haas/ext/switches/brocade.py

```diff
@@ -277,8 +277,18 @@ def _construct_tag(name):
         """ Construct the xml tag by prepending the brocade tag prefix. """
         return '{urn:brocade.com:mgmt:brocade-interface}%s' % name
 
-    def _make_request(self, method, url, data=None):
+    def _make_request(self, method, url, data=None, log=True):
         r = requests.request(method, url, data=data, auth=self._auth)
-        if r.status_code >= 400:
+
+        # error logging
+        # remove/delete something that's not attached/exists
+        if ((method == 'DELETE') and (not log) and (r.status_code == 404)):
```

Comment:
In that case I would have to do the checking and the logging in the caller method. Because we are already doing error logging in `_make_request` I thought it would be redundant to do it in the caller method. 

Any specific reason why you want to do it in the caller method though?